pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageCrypto.java

```diff
@@ -356,22 +358,37 @@ public boolean removeKeyCipher(String keyName) {
      * 
      * @return encryptedData if success
      */
-    public synchronized ByteBuf encrypt(ConcurrentOpenHashSet<String> encKeys, MessageMetadata.Builder msgMetadata, ByteBuf payload)
-            throws PulsarClientException {
+    public synchronized ByteBuf encrypt(ConcurrentOpenHashSet<String> encKeys, CryptoKeyReader keyReader,
+            MessageMetadata.Builder msgMetadata, ByteBuf payload) throws PulsarClientException {
 
         if (encKeys.isEmpty()) {
             return payload;
         }
 
         // Update message metadata with encrypted data key
-        encKeys.forEach(keyName -> {
-            if (encryptedDataKeyMap.get(keyName) != null) {
-                msgMetadata.addEncryptionKeys(KeyByteValue.newBuilder().setKey(keyName)
-                    .setValue(ByteString.copyFrom(encryptedDataKeyMap.get(keyName))).build());
+        List<String> keyNameList = encKeys.values();
```

Comment:
`encKeys.values()` creates a new list..instead we can use `forEach()` ?

Reply:
Its done to propagate exception above